eZ/Publish/API/Repository/Tests/Iterator/BatchIteratorTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Tests\Iterator;
+
+use eZ\Publish\API\Repository\Iterator\BatchIterator;
+use PHPUnit\Framework\TestCase;
+
+final class BatchIteratorTest extends TestCase
+{
+    public function testIterateOverDummyResultSet(): void
+    {
+        $expectedData = range(1, 100);
+
+        $iterator = new BatchIterator(new BatchIteratorTestAdapter($expectedData));
```

Comment:
Mock the adapter instead. We're interested in counting `fetch` calls as well, since you've got that optimization to prevent `fetch` calls if `batchSize` wasn't reached and Iterator became exhausted.

Reply:
Maybe something along the lines:

```php
$mock = $this->createMock(BatchIteratorAdapter::class);
$mock->method('fetch')->willReturnCallback(function ($offset, $limit): \Iterator {
    return function () use ($offset, $limit) {
        $realLimit = $offset + $limit;
        while ($offset < $realLimit) {
            yield $offset;
            $offset++;
        }
    }
})
```

So you get the iterator splitting for fetches without extra effort?

Reply:
Applied in https://github.com/ezsystems/ezplatform-kernel/pull/211/commits/a0df5d8c4e7a4f9e9d23f7a349aa150c45ab7f07 by introducing fetch coutner on test adapter